### ISSUE TITLE
Codegen

### ISSUE THREAD

**Issue by wilg** [hacker-cb/nanopb_cpp#5]
I'm a newbie to embedded programming, but this library seems close to what I need but I'm confused why there is so much boilerplate required. Can't all of this be auto-generated from the `.proto` file just like other protobuf implementations?

**Comment by wuyuanyi135** [comment]
Agreed, the code seems quite generatable and will bring a lot convinence to cpp development flow. I wonder if this feature would be considered?
